.github/workflows/upload-results.yml

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) Microsoft Corporation
+# SPDX-License-Identifier: MIT
+
+# This workflow executes a single test, optionally gathering code coverage and logs.
+
+name: Reusable Test Workflow
+
+on:
+  workflow_call:
+    inputs:
+      name:
+        required: true
+        type: string
+      result_artifact:
+        required: true
+        type: string
+
+jobs:
+  upload_results:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Download performance result artifacts
+      uses: actions/download-artifact@v2
+      with:
+        name: Test-Logs-${{inputs.result_artifact}}
+        path: ${{github.workspace}}/results_artifact
+
+    - name: Gather CSV results into results directory
+      run: |
+        mkdir -p ${{github.workspace}}/results/${{github.sha}}
+        curl https://raw.githubusercontent.com/microsoft/bpf_performance/1a50dae54c0c1d8b7f73331936e69a7154b3f659/scripts/gather_csv.sh > gather_csv.sh
```

Comment:
I think in another PR you were parametrizing the download to get the latest bits - is it easy to do it here too?

Reply:
The goal here is to make sure the script we get hasn't been altered.

Reply:
Sure, was just thinking in terms of not needing to update this line if there are changes to the script. Not blocking here

Reply:
That's intentional. If the script changes, then someone should review it before picking up the new version to make sure it's not malicious.